### PR TITLE
help: finish command help move, improve generic help

### PR DESCRIPTION
**Pull request by UniqMartin**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/master/.github/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes?
  - **No.** There's not much that can be tested. The changes are mostly in the documentation. (*Updated:* Added in #114.)
- [x] Have you successfully ran `brew tests` with your changes locally?

----

User-visible changes:

- `brew help <command>` now works for all commands documented in the man page.
- `--version` is now mostly treated like a regular command (e.g. it shows up under `brew commands` and is suggested by shell completions).
- `brew [help]` (the generic help text):
  - uses a more consistent styling,
  - ~~no longer lists the niche `pin` and `unpin` commands,~~
  - ~~drops `--env` in the “Troubleshooting” section,~~
  - mentions `brew help <command>` in the “Further help” section, and
  - ~~reorders the usage examples to better reflect a typical workflow (`search`, then query for details via `info` and friends, then `install`, later `upgrade` and/or `uninstall`). This is obviously subjective; I'm happy to hear other people's opinions here.~~

### CODE REVIEW

File: Library/Homebrew/cmd/help.rb [hunk 1]
```diff
@@ -21,6 +20,7 @@
 
 Further help:
   man brew
+  brew help COMMAND
```
Comment (mikemcquaid) [inline Library/Homebrew/cmd/help.rb]: `[COMMAND]`

File: Library/Homebrew/cmd/help.rb [hunk 2]
```diff
@@ -1,18 +1,17 @@
...
   brew list [FORMULA...]
   brew update
-  brew upgrade [FORMULA...]
-  brew pin/unpin [FORMULA...]
 
 Troubleshooting:
+  brew config
   brew doctor
   brew install -vd FORMULA
-  brew [--env | config]
```
Comment (mikemcquaid) [inline Library/Homebrew/cmd/help.rb]: Let's pull all this reordering into another PR for discussion.

**Comment by mikemcquaid** [comment]
A few comments otherwise 👍

File: Library/Homebrew/cmd/help.rb (hunk 1 above)
Comment (UniqMartin) [inline Library/Homebrew/cmd/help.rb]: My thinking was that this is the output of `brew help`, thus executing a plain `brew help` won't actually provide any “further help” (as the section title claims).

File: Library/Homebrew/cmd/help.rb (hunk 2 above)
Comment (UniqMartin) [inline Library/Homebrew/cmd/help.rb]: Agreed.

File: Library/Homebrew/cmd/help.rb (hunk 1 above)
Comment (mikemcquaid) [inline Library/Homebrew/cmd/help.rb]: They may have got this just from running `brew` so I think it's almost just a question of documentation of command arguments.

Comment (UniqMartin) [inline Library/Homebrew/cmd/help.rb]: And then they run `brew help` and get the exact same output (just a different exit code). But I don't mind adding the square brackets.

**Comment by UniqMartin** [comment]
Addressed code comments and updated the issue description to reflect these changes.

File: Library/Homebrew/cmd/help.rb (hunk 1 above)
Comment (mikemcquaid) [inline Library/Homebrew/cmd/help.rb]: 👍

**Comment by mikemcquaid** [comment]
Looks great 👍
